doc/src/async_example.rs

```diff
@@ -0,0 +1,76 @@
+// ANCHOR: all
+use calloop::{EventLoop, LoopSignal};
+
+// futures = "0.3"
+use futures::sink::SinkExt;
+use futures::stream::StreamExt;
+
+fn main() -> std::io::Result<()> {
+    // ANCHOR: decl_executor
+    let (exec, sched) = calloop::futures::executor()?;
+    // ANCHOR_END: decl_executor
+
+    // ANCHOR: decl_loop
+    // Just like in the timer example, we want to use a LoopSignal as the loop
+    // data so we can exit the program later.
+    let mut event_loop: EventLoop<LoopSignal> = EventLoop::try_new()?;
+    let handle = event_loop.handle();
+
+    handle.insert_source(exec, |evt, _metadata, loop_signaller| {
+        // Print the value of the async block ie. the return value.
+        println!("Async block ended with: {}", evt);
+        // Stop the loop when all blocks have run.
+        loop_signaller.stop();
```

Comment:
Doing so will actually stop the loop as soon as *any* future returns a value.

Reply:
> Doing so will actually stop the loop as soon as any future returns a value.

Oops. Begs the question of why I get the output that I do, but perhaps it's a timing thing. I'll push a fixed version.